apps/experiments/models.py

```diff
@@ -748,6 +748,23 @@ def compare_with_latest(self):
             version.compare(prev_version.version)
         return version.fields_changed
 
+    def archive(self):
+        super().archive()
+        self.delete_experiment_channels()
+
+    def delete_experiment_channels(self):
+        from apps.channels.models import ExperimentChannel
+
+        for channel in ExperimentChannel.objects.filter(experiment_id=self.id):
+            channel.soft_delete()
+
+    def archive_working_experiment(self):
+        if self.is_working_version:
+            if self.has_versions:
+                for version in self.versions.all():
+                    version.archive()
```

Comment:
I thought we decided we could do this with a query because we don't need to `delete_experiment_channels` for versions (since the channels are only attached to the working version.